src/jit/MemoryInl.h

```diff
@@ -173,6 +186,16 @@ void MemAddress::check(sljit_compiler* compiler, Operand* offsetOperand, sljit_u
             memArg.arg = SLJIT_MEM1(baseReg);
         }
 #endif /* SLJIT_32BIT_ARCHITECTURE */
+
+#if defined(ENABLE_EXTENDED_FEATURES)
+        if (options & CheckNaturalAlignment) {
+            if (SLJIT_IS_MEM2(memArg.arg)) {
+                sljit_emit_op2(compiler, SLJIT_ADD, TMP_REG1, 0, baseReg, 0, offsetReg, 0);
```

Comment:
Where does TMP_REG1 comes from? If you compute it, you could also overwrite  memArg.arg to SLJIT_MEM1